util/include/openssl_utils.hpp

```diff
@@ -0,0 +1,157 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+#pragma once
+
+#include <string>
+#include <memory>
+#include <fstream>
+
+#include <openssl/bio.h>
+#include <openssl/ec.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+#include <openssl/evp.h>
+
+#include "crypto_interface.hpp"
+#include "crypto_utils.hpp"
+
+using std::pair;
+using std::make_pair;
+using std::string;
+using std::unique_ptr;
+using std::ofstream;
+using concord::util::crypto::KeyFormat;
+using concord::util::cryptointerface::ISigner;
+using concord::util::cryptointerface::IVerifier;
+
+namespace concord::util::openssl_utils {
+
+constexpr static size_t SIG_LENGTH{64};
+
+// Deleter classes.
+class EVP_PKEY_Deleter {
+ public:
+  void operator()(EVP_PKEY* p) const { EVP_PKEY_free(p); }
+};
+
+class EVP_MD_CTX_Deleter {
+ public:
+  void operator()(EVP_MD_CTX* p) const { EVP_MD_CTX_free(p); }
+};
+
+class EVP_PKEY_CTX_Deleter {
+ public:
+  void operator()(EVP_PKEY_CTX* p) const { EVP_PKEY_CTX_free(p); }
+};
+
+class CertificateUtils {
+ public:
+  static std::string generateSelfSignedCert(const std::string& origin_cert_path,
+                                            const std::string& pub_key,
+                                            const std::string& signing_key);
+  static bool verifyCertificate(X509* cert, const std::string& pub_key);
+  static bool verifyCertificate(X509* cert_to_verify,
+                                const std::string& cert_root_directory,
+                                uint32_t& remote_peer_id,
+                                std::string& conn_type,
+                                bool use_unified_certs);
+};
+
+// This class implements OpenSSL's EdDSA signer.
+class EdDSA_Signer : public ISigner {
```

Comment:
Isn't using underscores in class names against the conventions?
Shouldn't this be implemented in a separate file?
We need to talk about synchronizing my implementation and yours.

Reply:
Changed to `EdDSASigner`, `EdDSAVerifier`.